pkg/controller/metrics/metrics.go

```diff
@@ -645,8 +650,18 @@ func (ca *clusterAccumulator) processCluster(cd *hivev1.ClusterDeployment) {
 func (ca *clusterAccumulator) setMetrics(total, installed, uninstalled, deprovisioning, conditions *prometheus.GaugeVec, mcLog log.FieldLogger) {
 
 	for k, v := range ca.total {
-		total.WithLabelValues(k, ca.ageFilter).Set(float64(v))
+		for clusterType := range ca.clusterTypesSet {
+			if count, ok := v[clusterType]; ok {
+				total.WithLabelValues(clusterType, ca.ageFilter, string(k)).Set(float64(count))
+			} else {
+				// We need to potentially clear out old cluster types no longer showing in the list.
+				// This will work so long as there is at least one cluster of that type still remaining
+				// in hive somewhere.
+				total.WithLabelValues(clusterType, ca.ageFilter, string(k)).Set(float64(0))
+			}
```

Comment:
I see where you're following the pattern of the existing metrics, but I think we might not need this conditional, since a map returns the zero value of the member type when the key doesn't exist.
```suggestion
			total.WithLabelValues(clusterType, ca.ageFilter, string(k)).Set(float64(v[clusterType]))
```

That said, there's no need to "fix" it here -- may as well roll them all up as tech debt for later.

It's also vaguely disturbing that, if we delete the last cluster of a given `clusterType`, its count will remain whatever the last reported value was. But again, that's latent and need not be addressed here.

Reply:
+1 to Eric's suggested change, but the reason is - [this](https://github.com/openshift/hive/blob/master/pkg/controller/metrics/metrics.go#L193:L196) ensures [Start method](https://github.com/openshift/hive/blob/master/pkg/controller/metrics/metrics.go#L220) is run every 2 minutes, so the values are going to be reset anyways :)